server/handlers/validation.go

```diff
@@ -127,9 +126,67 @@ func validateUpdate(cs signed.CryptoService, gun string, updates []storage.MetaU
 		if err != nil {
 			return nil, err
 		}
-		updates = append(updates, *update)
+		updatesToApply = append(updatesToApply, *update)
+	}
+	return updatesToApply, nil
+}
+
+func loadAndValidateTargets(gun string, repo *tuf.Repo, roles map[string]storage.MetaUpdate, kdb *keys.KeyDB, store storage.MetaStore) ([]storage.MetaUpdate, error) {
+	targetsRoles := make(utils.RoleList, 0)
+	for role := range roles {
+		if role == data.RoleName(data.CanonicalTargetsRole) || data.IsDelegation(role) {
+			targetsRoles = append(targetsRoles, role)
+		}
+	}
+
+	// by sorting, we'll always process shallower targets updates before deeper
```

Comment:
should be: by sorting by segment, we'll...

Doesn't make sense when the first thought of the reader is "alphabetic sorting"